src/pymor/models/basic.py

```diff
@@ -126,6 +125,7 @@ def __init__(self, operator, rhs, outputs=None, products=None,
         self.rhs = rhs
         self.outputs = FrozenDict(outputs or {})
         self.solution_space = self.operator.source
+        self.linear = operator.linear and all(output.linear for output in outputs)
```

Comment:
`outputs` is a dict so should be

`all(output.linear for output in outputs.values())`

Reply:
Of course. I've changed it to `self.outputs.values()` to not check for `outputs is None`.